### ISSUE TITLE
A question about initial learning rate.

### ISSUE THREAD

**Issue by MenghaoGuo** [qq456cvb/Point-Transformers#3]
Thanks for your attention to PCT !
In our experiments, I find the PCT is sensitive to initial lr. 1e-3 is too large to make pct converge. I change the initial lr from 1e-3 to 1e-4 and get following results.

batch_size: 16
epoch: 200
gpu: 1
learning_rate: 0.0001
model:
  name: Menghao
name: Menghao
normal: true
num_point: 1024
optimizer: Adam
weight_decay: 0.0001

[2021-01-13 11:59:39,205][__main__][INFO] - Load dataset ...
The size of train data is 9843
The size of test data is 2468
[2021-01-13 11:59:42,420][__main__][INFO] - No existing model, starting training from scratch...
[2021-01-13 11:59:42,421][__main__][INFO] - Start training...
[2021-01-13 11:59:42,421][__main__][INFO] - Epoch 1 (1/200):
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 616/616 [03:09<00:00,  3.25it/s]
[2021-01-13 12:02:51,857][__main__][INFO] - Train Instance Accuracy: 0.387277
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 155/155 [00:48<00:00,  3.20it/s]
[2021-01-13 12:03:40,472][__main__][INFO] - Test Instance Accuracy: 0.511694, Class Accuracy: 0.345986
[2021-01-13 12:03:40,472][__main__][INFO] - Best Instance Accuracy: 0.511694, Class Accuracy: 0.345986
[2021-01-13 12:03:40,472][__main__][INFO] - Save model...
[2021-01-13 12:03:40,472][__main__][INFO] - Saving at best_model.pth
[2021-01-13 12:03:40,559][__main__][INFO] - Epoch 2 (2/200):
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 616/616 [03:13<00:00,  3.18it/s]
[2021-01-13 12:06:54,418][__main__][INFO] - Train Instance Accuracy: 0.488636
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 155/155 [00:48<00:00,  3.16it/s]
[2021-01-13 12:07:43,537][__main__][INFO] - Test Instance Accuracy: 0.663710, Class Accuracy: 0.506854
[2021-01-13 12:07:43,538][__main__][INFO] - Best Instance Accuracy: 0.663710, Class Accuracy: 0.506854
[2021-01-13 12:07:43,538][__main__][INFO] - Save model...
[2021-01-13 12:07:43,538][__main__][INFO] - Saving at best_model.pth
[2021-01-13 12:07:43,680][__main__][INFO] - Epoch 3 (3/200):
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 616/616 [03:09<00:00,  3.25it/s]
[2021-01-13 12:10:53,202][__main__][INFO] - Train Instance Accuracy: 0.549682
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 155/155 [00:49<00:00,  3.11it/s]
[2021-01-13 12:11:43,145][__main__][INFO] - Test Instance Accuracy: 0.747581, Class Accuracy: 0.612702
[2021-01-13 12:11:43,146][__main__][INFO] - Best Instance Accuracy: 0.747581, Class Accuracy: 0.612702
[2021-01-13 12:11:43,146][__main__][INFO] - Save model...
[2021-01-13 12:11:43,146][__main__][INFO] - Saving at best_model.pth
[2021-01-13 12:11:43,239][__main__][INFO] - Epoch 4 (4/200):
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 616/616 [03:11<00:00,  3.22it/s]
[2021-01-13 12:14:54,499][__main__][INFO] - Train Instance Accuracy: 0.592549
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 155/155 [00:49<00:00,  3.16it/s]
[2021-01-13 12:15:43,646][__main__][INFO] - Test Instance Accuracy: 0.799597, Class Accuracy: 0.684245
[2021-01-13 12:15:43,647][__main__][INFO] - Best Instance Accuracy: 0.799597, Class Accuracy: 0.684245
[2021-01-13 12:15:43,647][__main__][INFO] - Save model...
[2021-01-13 12:15:43,647][__main__][INFO] - Saving at best_model.pth
[2021-01-13 12:15:43,773][__main__][INFO] - Epoch 5 (5/200):
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 616/616 [03:10<00:00,  3.23it/s]
[2021-01-13 12:18:54,845][__main__][INFO] - Train Instance Accuracy: 0.625095
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 155/155 [00:48<00:00,  3.19it/s]
[2021-01-13 12:19:43,585][__main__][INFO] - Test Instance Accuracy: 0.813306, Class Accuracy: 0.711771
[2021-01-13 12:19:43,586][__main__][INFO] - Best Instance Accuracy: 0.813306, Class Accuracy: 0.711771
[2021-01-13 12:19:43,586][__main__][INFO] - Save model...
[2021-01-13 12:19:43,586][__main__][INFO] - Saving at best_model.pth
[2021-01-13 12:19:43,692][__main__][INFO] - Epoch 6 (6/200):
  7%|██████████▏                                                                                                                                    | 44/616 [00:15<02:27,  3.89it/s]

Best Regards,
Menghao

**Comment by MenghaoGuo** [comment]
In the end, the accuracy I got was 92.6.

[2021-01-14 01:53:36,615][__main__][INFO] - Test Instance Accuracy: 0.918952, Class Accuracy: 0.882602
[2021-01-14 01:53:36,616][__main__][INFO] - Best Instance Accuracy: 0.925984, Class Accuracy: 0.896355
[2021-01-14 01:53:36,616][__main__][INFO] - End of training...

**Comment by qq456cvb** [comment]
Thanks for your suggestion and glad to see such an improvement! I will update this result in README.

**Comment by MenghaoGuo** [comment]
Thanks.
